Header header
string current_state
string[] possible_transitions
string[] possible_transition_states